app/flatpak-builtins-build-export.c

```diff
@@ -294,6 +294,157 @@ find_file_in_tree (GFile *base, const char *filename)
   return FALSE;
 }
 
+typedef gboolean (* VisitFileFunc) (GFile *file, GError **error);
+
+static gboolean
+visit_files_in_tree (GFile *base,
+                     VisitFileFunc visit_file,
+                     gpointer data)
+{
+  g_autoptr(GFileEnumerator) enumerator = NULL;
+  GError **error = data;
+
+  enumerator = g_file_enumerate_children (base,
+                                          G_FILE_ATTRIBUTE_STANDARD_TYPE ","
+                                          G_FILE_ATTRIBUTE_STANDARD_NAME,
+                                          G_FILE_QUERY_INFO_NONE,
+                                          NULL,
+                                          NULL);
+  if (!enumerator)
+    return FALSE;
+
+  do
+    {
+      GFileInfo *info;
+      GFile *child;
+      GFileType type;
+
+      if (!g_file_enumerator_iterate (enumerator, &info, &child, NULL, error))
+        return FALSE;
+
+      if (!info)
+        return TRUE;
+
+      type = g_file_info_get_file_type (info);
+
+      if (type == G_FILE_TYPE_REGULAR)
+        {
+          if (!visit_file (child, data))
+            return FALSE;
+        }
+      else if (type == G_FILE_TYPE_DIRECTORY)
+        {
+          if (!visit_files_in_tree (child, visit_file, data))
+            return FALSE;
+        }
+    }
+  while (1);
+
+  return TRUE;
+}
+
+G_GNUC_NULL_TERMINATED
+static void
+add_args (GPtrArray *argv_array, ...)
+{
+  va_list args;
+  const char *arg;
+
+  va_start (args, argv_array);
+  while ((arg = va_arg (args, const gchar *)))
+    g_ptr_array_add (argv_array, g_strdup (arg));
+  va_end (args);
+}
+
+static void
+add_env (GPtrArray  *array,
+         const char *envvar)
+{
+  if (g_getenv (envvar) != NULL)
+    add_args (array,
+              "--setenv", envvar, g_getenv (envvar),
+              NULL);
+}
+
+static void
+add_bwrap (GPtrArray *array,
+           const char *input)
+{
+  add_args (array,
+            flatpak_get_bwrap (),
+            "--ro-bind", "/usr", "/usr",
+            "--ro-bind", "/lib", "/lib",
+            "--ro-bind", "/lib64", "/lib64",
```

Comment:
This will fail if the system is not using /lib64.

Reply:
Also, i think we need /etc for things like ld.so.*
I think it might be easier to do a full expose of / like we do in the triggers instead: https://github.com/flatpak/flatpak/blob/master/common/flatpak-dir.c#L5648

Also, like that we should probably respect DISABLE_SANDBOXED_TRIGGERS so that it keeps working inside a flatpak.